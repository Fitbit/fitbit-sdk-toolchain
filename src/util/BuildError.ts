import ErrorSubclass from 'error-subclass';

import {
  Diagnostic,
  DiagnosticCategory,
  DiagnosticTarget,
} from '../diagnostics';

export default class BuildError extends ErrorSubclass {
  static displayName = 'BuildError';

  target?: DiagnosticTarget;

  static is(error: Error): error is BuildError {
    return error instanceof BuildError;
  }

  toDiagnostic(): Diagnostic {
    return {
      category: DiagnosticCategory.Error,
      messageText: this.message,
      ...(this.target && { target: this.target }),
    };
  }
}
