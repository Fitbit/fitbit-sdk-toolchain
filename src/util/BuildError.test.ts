import BuildError from './BuildError';

describe('is()', () => {
    it('returns false if error is not a BuildError', () => expect(BuildError.is(new Error())).toBe(false));

    it('returns true if error is a BuildError', () => expect(BuildError.is(new BuildError())).toBe(true));
});

describe('toDiagnostic()', () => {
    it('returns an error diagnostic', () => expect(new BuildError('uh oh!').toDiagnostic()).toMatchSnapshot());
});
