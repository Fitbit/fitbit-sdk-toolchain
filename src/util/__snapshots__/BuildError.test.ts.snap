// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toDiagnostic() returns an error diagnostic 1`] = `
Object {
  "category": 1,
  "messageText": "uh oh!",
}
`;
